src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -202,6 +206,14 @@ protected override void ExecuteCore()
                             string runtimePackName = runtimePackNamePattern.Replace("**RID**", runtimeIdentifier);
                             preferredPackages.Add(runtimePackName);
                         }
+
+                        // Update the known runtime identifier platforms based on the selected Microsoft.NETCore.App pack
+                        if (selectedRuntimePack?.Name == "Microsoft.NETCore.App")
```

Comment:
It's probably better to do a case-insensitive comparison here.

```suggestion
                        if (selectedRuntimePack?.Name?.Equals("Microsoft.NETCore.App", StringComparison.InvariantIgnoreCase))
```